src/utilities/data/Vector.cpp

```diff
@@ -92,16 +96,28 @@ InterpInfo interpInfo(const Vector& x, double xi) {
 
     result.ia = (unsigned)(it - begin - 1);
     result.ib = (unsigned)(it - begin);
-    result.wa = (x(result.ib) - xi) / (x(result.ib) - x(result.ia));
-    result.wb = (xi - x(result.ia)) / (x(result.ib) - x(result.ia));
+    if (ti == 0.0) {
+      result.wa = (x(result.ib) - xi) / (x(result.ib) - x(result.ia));
+      result.wb = (xi - x(result.ia)) / (x(result.ib) - x(result.ia));
+    } else {
+      result.wb = xi - x(result.ia);
+      result.wa = ti - result.wb;
+      if ((result.wb > 0.0) && (result.wb < ti)) {
+        result.ti = ti;
+      } else {
+        result.wa = 0.0;
+        result.wb = 1.0;
+        result.ti = 1.0;
+      }
+    }
   }
 
   return result;
 }
 
 /// linear interpolation of the function y = f(x) at point xi
 /// assumes that x is strictly increasing
-double interp(const Vector& x, const Vector& y, double xi, InterpMethod interpMethod, ExtrapMethod extrapMethod) {
+double interp(const Vector& x, const Vector& y, double xi, InterpMethod interpMethod, ExtrapMethod extrapMethod, double ti) {
```

Comment:
I'd like to see an assertion that `ti` must be valid when interpMethod is AverageInterp (I think that's the only one where it's needed)